alacritty_terminal/src/term/mod.rs

```diff
@@ -1763,6 +1767,12 @@ impl<T: EventListener> Handler for Term<T> {
                 }
             },
         }
+
+        let buffer_cursor_line = self.grid.buffer_cursor_point().line;
+        self.selection = self
+            .selection
+            .take()
+            .filter(|s| !s.intersects_range(buffer_cursor_line..=buffer_cursor_line));
```

Comment:
In theory an `intersects_line` function like you had previously would make this only a single line.

My main problem was with duplication of code, do you think it's worth it to have a method that basically just calls `self.intersects_range(line..=line)` internally?

Formatting is maybe not the best reason to do so, just an idea.

Reply:
maybe, but I don't see how it could improve anything, since 1..=1 is still a valid range, so I don't see any issue. So, I'd prefer to keep it that way.